src/bin/cargo-fixit/fixit.rs

```diff
@@ -162,15 +170,37 @@ fn run_rustfix(
         }
 
         file_map
+            .entry(package_id)
+            .or_insert_with(IndexMap::new)
             .entry(file_name)
             .or_insert_with(IndexSet::new)
             .insert((suggestion, diagnostic.rendered));
     }
 
     let _exit_code = command.wait()?;
 
+    let Some(package_id) = pkg_id
+        .clone()
+        .and_then(|p| {
+            if file_map.contains_key(&p) {
+                Some(p)
+            } else {
+                None
+            }
+        })
+        .or_else(|| file_map.first_entry().map(|entry| entry.key().clone()))
+    else {
+        *pkg_id = None;
+        return Ok((errors, false));
+    };
+
+    let Some(file_suggestions) = file_map.get(&package_id) else {
+        *pkg_id = None;
+        return Ok((errors, false));
+    };
```

Comment:
This feels a bit redundant, checking the map for an id and then lookup up the id.